x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/Security.java

```diff
@@ -402,31 +375,13 @@ protected Clock getClock() {
         securityContext.set(new SecurityContext(settings, threadPool.getThreadContext()));
         components.add(securityContext.get());
 
-        // audit trails construction
-        Set<AuditTrail> auditTrails = new LinkedHashSet<>();
+        // audit trail service construction
+        final List<AuditTrail> auditTrails = new ArrayList<>();
```

Comment:
no need to allocate a list for this. Just do something like:

```suggestion
        final List<AuditTrail> auditTrails = XPackSettings.AUDIT_ENABLED.get(settings) ? 
            Collections.singletonList(new LoggingAuditTrail(settings, clusterService, threadPool)) : 
            Collections.emptyList();
```

Note: this is probably poorly formatted.